frontend/public/components/utils/container-select.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import {
+  Divider,
+  MenuToggle,
+  MenuToggleElement,
+  Select,
+  SelectGroup,
+  SelectList,
+  SelectOption,
+} from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+
+import { ContainerModel } from '@console/internal/models';
+import { ContainerSpec } from '@console/internal/module/k8s';
+import { ResourceName } from './resource-icon';
+
+export const ContainerLabel: React.FC<ContainerLabelProps> = ({ name }) => (
+  <ResourceName name={name} kind={ContainerModel.kind} />
+);
+
+const ContainerSelectOptions: React.FC<ContainerSelectOptionsProps> = ({ containers }) => (
+  <>
+    {Object.values(containers).map(({ name }) => (
+      <SelectOption key={name} value={name}>
+        <ContainerLabel name={name} />
+      </SelectOption>
+    ))}
+  </>
+);
+
+export const ContainerSelect: React.FC<ContainerSelectProps> = ({
+  containers,
+  currentKey,
+  initContainers,
+  onChange,
+}) => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [selected, setSelected] = React.useState<string>(
+    currentKey || Object.values(containers)[0].name,
```

Comment:
We probably need optional chaining here. This could cause an RTE if containers is undefined or an empty objecct
```suggestion
    currentKey || Object.values(containers ?? {}).?[0]?.name,
```